detectable/src/main/java/com/synopsys/integration/detectable/detectables/maven/cli/MavenCliExtractor.java

```diff
@@ -46,13 +49,27 @@ public Extraction extract(File directory, ExecutableTarget mavenExe, MavenCliExt
         List<String> includedScopes = mavenCliExtractorOptions.getMavenIncludedScopes();
         List<String> excludedModules = mavenCliExtractorOptions.getMavenExcludedModules();
         List<String> includedModules = mavenCliExtractorOptions.getMavenIncludedModules();
+        Boolean includeShadedDependencies = mavenCliExtractorOptions.getMavenIncludeShadedDependencies();
+
+        if(includeShadedDependencies) {
+            try {
+                mavenProjectInspectorDetectable.includeShadedDependencies = "include_shaded_dependencies";
+                mavenProjectInspectorDetectable.extractable();
+                Extraction extraction = mavenProjectInspectorDetectable.extract(extractionEnvironment);
```

Comment:
I can see how it is tempting to pull in the whole MavenProjectInspectorDetectable as a dependency; however, it seems to me that since Detectable is such a high-level purpose-specific concept it would be better to extract whatever is needed into another class.

We can get some feedback from the team on this as well.

Reply:
I was having the same doubts as whether I should create a new class for getting the PI executable for getting just the shaded dependencies, but doing so it would cause a lot of complex code and code repetition as we want all the methods and classes which this Detectable has to correctly execute PI.

In an ideal case, the new class would go under the DetectorRuleFactory to follow the cascading rules and under there we would execute PI, but this is an exceptional scenario where we want to execute PI in build mode which is why we need the Detectable under this class. A new Detectable would have the exact same features as MavenProjectInspectorDetectable so I decided to go with the existing one but if the team feels that this deserves an extra class, I would make those changes.

Reply:
Thanks, Dev. Just to clarify, I don't think we need a "new detectable".
The functionality can be in a different class that both Detectables use.
Alternatively, we could have a base class which both delectables extend.

With this current change we have a dependency path which is something like this:
`DetectableA -> ExtractorA -> DetectableB -> ExtractorB -> PIParser`.

I'll leave it as this and let another reviewer chime in. :)